src/System.IO.Compression.Brotli/README.md

```diff
@@ -0,0 +1,219 @@
+        public static int GetMaximumCompressedSize(int inputSize);
+        public struct State : IDisposable {
+            public bool CompressMode { get; }
+            public void Dispose();
+            public void SetQuality();
+            public void SetQuality(uint quality);
+            public void SetWindow();
+            public void SetWindow(uint window);
+        }
+    }
+```
+```out bytesConsumed``` - number of bytes from source, which used in this call
+
+```out bytesWritten``` - number of written to destination bytes 
+
+The ```Compress``` performs data compression. It sets ```source``` data should be compressed and return compressed data in ```destination```. ```state.CompressMode``` should be ```true``` .
+
+
+The ```FlushEncoder``` return compressed data, which have sent to ```state``` using Compress. 
+
+The ```Decompress``` Decompresses the data in ```source``` into ```destination```. ```state.CompressMode``` should be ```false``` .
+
+```State``` uses in both compress and decompress mode to save a temporary status and data of process.
+
+```CompressMode =  true``` shows that ```State``` in compress mode, ```false``` in decompress mode. 
+
+```SetQuality``` alows you to set quality of compression ```0 to 11```. The higher quality means higher compression ratio. 
+
+```SetWindow``` - Logarithm of Recommended sliding LZ77 window size. Encoder may reduce this value, e.g. if input is much smaller than window size. Window size is (1 << value) - 16. Possible values: ```11 to 24```
+
+### Examples
+Simple method to compress bytes to file.
+```C#
+
+ static void CompressSimple(byte[] bytes, string OutFile)
+ {
+    Brotli.State state = new Brotli.State();
+    byte[] compressed = new byte[Brotli.GetMaximumCompressedSize(bytes.Length)];
+    state.SetQuality();
+    TransformationStatus result = Brotli.Compress(bytes, compressed, out int consumed, out int written, ref state);
+    while (result != TransformationStatus.Done)
+    {
+        result = Brotli.Compress(bytes, compressed, out consumed, out written, ref state);
+        bytes = bytes.AsSpan().Slice(consumed).ToArray();
+    }
+    result = Brotli.FlushEncoder(Array.Empty<byte>(), compressed, out consumed, out written, ref state);
+    byte[] ans = new byte[written];
+    Array.Copy(compressed, ans, written);
+    File.WriteAllBytes(OutFile, ans);
+ }
+```
+
+Simple method to decompress bytes to file. If out data is larger than destination ```result``` be ```TransformationStatus.DestinationTooSmall``` and it's available to call Decompress again to collect rest of data.
+
+```C#
+ static void DecompSimple(byte[] bytes, string outFile, int decompressedLength)
+ {
+    Brotli.State state = new Brotli.State();
+    byte[] data = ms.ToArray();
+    byte[] decompressed = new byte[decLen];
+    TransformationStatus result = Brotli.Decompress(data, decompressed, out int consumed, out int written, ref state);
```

Comment:
FlushEncode is only for compression, because we should say that we won't sent any more data for compress and want to get compressed data.